cmd/posix-errors.go

```diff
@@ -80,8 +75,7 @@ func isSysErrIsDir(err error) bool {
 // Check if the given error corresponds to ENOTDIR (is not a directory).
 func isSysErrNotDir(err error) bool {
 	if pathErr, ok := err.(*os.PathError); ok {
-		switch pathErr.Err {
-		case syscall.ENOTDIR:
+		if pathErr.Err == syscall.ENOTDIR {
```

Comment:
Further simplification is possible simply do `return pathErr.Err == syscall.ENOTDIR`